packages/hash/frontend/src/blocks/page/focusPageTitlePlugin.ts

```diff
@@ -0,0 +1,27 @@
+import { Schema } from "prosemirror-model";
+import { Plugin } from "prosemirror-state";
+
+export const focusPageTitlePlugin = new Plugin<any, Schema>({
+  props: {
+    handleKeyDown: (view, event) => {
+      const isArrowUp = event.key === "ArrowUp";
+
+      if (!isArrowUp) return false;
+
+      const { selection } = view.state;
+      const posAtIndex = selection.$head.posAtIndex(0, 1);
+
+      /** @todo `posAtIndex === 1` only works if first item is a text block, fix this */
+      if (posAtIndex === 1) {
+        const pageTitle =
+          document.querySelector<HTMLHeadingElement>("h1#hash-page-title");
```

Comment:
But this code is in a plugin, so we cannot call hooks to consume context here.

Reply:
You're right, my bad. There's actually another possible way to do this by passing it through createEditorView in PageBlock, but if we're eventually going to turn the title into a block inside Prosemirror I think we can skip this.

Reply:
I agree with @luisbettencourt that we should avoid `querySelector` – it wouldn't take long to pass a `focusPageTitle` function into a `createFocusPageTitlePlugin` function from `createEditorView` from `PageBlock` and from its parent.

Reply:
Is there a specific reason or concern for not using `querySelector`, but do a prop passing chain like you suggested? I'm ok with both, but trying to see the reason behind choosing one to another.

Reply:
It just breaks React's best practices so we avoid using it where we can. There are some default benefits that come with using refs instead of querying the DOM directly (code resiliency being one of them).

I think ideally we wouldn't use it at all, but Prosemirror is a thing of itself inside of a React project, so some compromises are made.

Reply:
Used ref instead of querySelector here https://github.com/hashintel/hash/pull/956/commits/86d923a0babd04959f24caa4e3a6bf14d28c5a8f